### PR TITLE
Add beta1 release notes

### LINKED ISSUE #26: Incorporate release notes into the official ecctl documentation
## Overview

We pushed some release notes for 1.0.0-rc1 in https://github.com/elastic/ecctl/releases/tag/1.0.0-rc1, but our standard is to include release notes with the official Asciidoctor-sourced documentation.  

## Possible Implementation

The release notes currently are actually a changelog; we typically try to curate the most important changes and bug fixes into the release highlights. 

We should discuss if we want to add the release notes to the official docs and remove them from GitHub or if this information can be duplicated. (For ESS and ECE, the only release notes we provide are part of the official docs, which is also true for many other Elastic products.)

## Testing

We would build and proof the release notes content as part of our regular docs release activities.

### PR DESCRIPTION
**Pull request by nrichers**
## Description

This PR adds release notes for 1.0.0-beta1 in a new "Release notes" section to our official ecctl documentation.

Content is based on https://docs.google.com/document/d/1nUq8cCiuMingMxYU-GvPojyuDi94jXieuh_oMonQzd0/edit# with some edits.

## Related Issues

Closes #26 

## How Has This Been Tested?

Content builds cleanly. Output:

image

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Readiness Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: docs/ecctl-release-notes.asciidoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+[id="{p}-release-notes"]
+== Release notes
+
+This section summarizes the changes in each {n} release.
+
+* <<{p}-release-notes-1.0.0-beta1>>
+
+[id="{p}-release-notes-1.0.0-beta1"]
+=== Elastic Cloud Control (ecctl) 1.0.0-beta1
```
Comment (karencfv) [inline docs/ecctl-release-notes.asciidoc]:
```suggestion
=== Elastic Cloud Control (ecctl) v1.0.0-beta1
```

File: docs/ecctl-release-notes.asciidoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+++++
+<titleabbrev>1.0.0-beta1</titleabbrev>
+++++
+
+Elastic cloud control (ecctl) is Elastic’s CLI interface to manage the Elastic Cloud Enterprise platform and is now open for a public beta.
+
+New for 1.0.0-beta1:
+
+// TODO What are the ecctl creation commands? When I run `ecctl` (not on beta1, mind), I get auth, deployment, generate, help, init, platform, user, and version. The `ecctl deployment` command maybe? You can do an ecctl deployment elasticsearch create, for example.
```
Comment (karencfv) [inline docs/ecctl-release-notes.asciidoc]: The creation commands are the old `ecctl deployment elasticsearch create` command (which has now been removed in favour of `ecctl deployment create`) and the current `ecctl deployment kibana create` and `ecctl deployment apm create` commands.

File: docs/ecctl-release-notes.asciidoc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+++++
+
+Elastic cloud control (ecctl) is Elastic’s CLI interface to manage the Elastic Cloud Enterprise platform and is now open for a public beta.
+
+New for 1.0.0-beta1:
+
+// TODO What are the ecctl creation commands? When I run `ecctl` (not on beta1, mind), I get auth, deployment, generate, help, init, platform, user, and version. The `ecctl deployment` command maybe? You can do an ecctl deployment elasticsearch create, for example. 
+* *New deployment APIs for ecctl creation commands*. With the availability of our deployment APIs for ECE, we’ve switched the ecctl creation commands to make use of the new APIs. This switch ensures better supportability and compatibility with ECE functionality and allows you to fully control your deployments. 
++
+Use the ecctl creation commands to:
```
Comment (karencfv) [inline docs/ecctl-release-notes.asciidoc]:
```suggestion
Use the ecctl deployment commands to:
```

Comment (nrichers) [reply docs/ecctl-release-notes.asciidoc]: Updated gobally to refer to the `ecctl deployment` commands, thank you!

File: docs/ecctl-release-notes.asciidoc (hunk 2 above)
Comment (nrichers) [reply docs/ecctl-release-notes.asciidoc]: Thank you for explaining! I've updated the release notes as per your other comment.

File: docs/ecctl-release-notes.asciidoc (hunk 1 above)
Comment (nrichers) [reply docs/ecctl-release-notes.asciidoc]: @karencfv I've updated the release notes globally as per your comment. I see that `v1.0.0-beta1` is the tag you are using for this release, but it's somewhat different from the convention that other products use at Elastic. ECE was `1.0.0-beta1`, for example, no `v` prefix.

**Comment by nrichers** [comment]
> This is looking great, thank you! Should we also mention the new deployment resource commands and the breaking changes?

@karencfv can you suggest some information for both of these? I don't have any information to go by. (I can try out the deployment resource commands a bit to see if I can figure out that bit.)

It would also be helpful if we could tell people how to upgrade or migrate to v1.0.0-beta1. Do they just need to reinstall?

**Comment by Kushmaro** [comment]
Thanks @nrichers , regarding breaking changes, I'm not sure if we should mention those as the tool was really just available in beta mode up until this point, but there was no official release.

I think that if we commit further breaking changes from beta->GA we should announce, but for now we can skip that (the tool is only 'ready' technically just now, in this release)

**Comment by marclop** [comment]
@nrichers The changelog which I generated can be found here: https://github.com/elastic/ecctl/releases/tag/v1.0.0-beta1

File: docs/ecctl-release-notes.asciidoc (hunk 1 above)
Comment (karencfv) [reply docs/ecctl-release-notes.asciidoc]: ah! thanks for that information. I guess this is something that we might need to change in the future.

**Pull request merged**